### ISSUE TITLE
Symbolic links in link mess (for home directory) gives errors

### ISSUE THREAD

**Issue by pontus** [apptainer/singularity#4836]
### Version of Singularity:

What version of Singularity are you using? Run:

```
singularity version 3.5.1-1.el7
```

### Expected behavior

What did you expect to see when you do...?

```
dahlo@snowy-lr2 ~ $ ls -ld ~/work/testarea
lrwxrwxrwx 1 dahlo dahlo 45 19 apr  2018 /home/dahlo/work/testarea -> /proj/sllstore2017094/nobackup/dahlo/testarea
dahlo@snowy-lr2 ~ $ cd /home/dahlo/work/testarea/temp/
dahlo@snowy-lr2 ~/work/testarea/temp $ singularity shell ubuntu.18.04.sif 
```

### Actual behavior

What actually happend? Why was it incorrect?

We get
```
FATAL:   container creation failed: hook function for tag sessiondir returns error: failed to read symlink /home/dahlo/work/testarea/temp: readlink /home/dahlo/work/testarea/temp: invalid argument
```

instead of the container starting.

### When it doesn't break

If instead we do:

```
dahlo@snowy-lr2 ~/work/testarea/temp $ pwd -P
/crex/proj/sllstore2017094/nobackup/dahlo/testarea/temp
dahlo@snowy-lr2 ~/work/testarea/temp $ cd /crex/proj/sllstore2017094/nobackup/dahlo/testarea/temp
dahlo@snowy-lr2 /crex/proj/sllstore2017094/nobackup/dahlo/testarea/temp $ singularity shell ubuntu.18.04.sif 
Singularity> exit
exit
dahlo@snowy-lr2 /crex/proj/sllstore2017094/nobackup/dahlo/testarea/temp $ 
```

things work as expected.

Also if we skip the last part in CWD:

```
dahlo@snowy-lr2 ~ $ cd /home/dahlo/work/testarea
dahlo@snowy-lr2 ~/work/testarea $ singularity shell temp/ubuntu.18.04.sif 
Singularity> exit
exit
dahlo@snowy-lr2 ~/work/testarea $ 
```
It also seems to be a requirement to trigger this that the link is under the home directory.

### Steps to reproduce this behavior

How can others reproduce this issue/problem?

Unclear as of yet.

### What OS/distro are you running

```

dahlo@snowy-lr2 ~ $ cat /etc/os-release 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

dahlo@snowy-lr2 ~ $ 


```

### How did you install Singularity

Locally built rpm from source tarball.

### Debug output

Debug info for a failed run:

[dahlo_singularity_error_debug.txt](https://github.com/sylabs/singularity/files/3931543/dahlo_singularity_error_debug.txt)

**Comment by cclerget** [comment]
Hi @pontus , I will remove the corresponding error as it means there is an existing directory/file so the error is useless.

**Comment by pontus** [comment]
Hi, I'm not sure I understand, do you mean this is just due to an unnecessary fatal/die or similar?

**Comment by cclerget** [comment]
Exactly, here the fix #4838 if you can give it a try and confirm it works. Thanks !

**Comment by pontus** [comment]
I can confirm this fixes the issue without any observed ill effects.
